### DISCUSSION TITLE
Error message on npg_helpers

### DISCUSSION THREAD

**Discussion by cjchristensen8** [Dan-Patterson/npgeom#2] [General]
hi Dan - a huge thanks for craeting this toolset, and i offer my great admiration for your help to the community.  
When I tried the tool to rotate objects, I got this error string (using ArcGIS Pro 2.5 on a Windows 10 machine):

 #########################
Traceback (most recent call last):
  File "C:\Users\*******\Documents\ArcGIS\SDK\numpy_geometry-master\arcpro_npg\npg\tbx_tools.py", line 131, in <module>
    from npg.npg_create import circle
  File "C:\Users\*******\Documents\ArcGIS\SDK\numpy_geometry-master\arcpro_npg\npg\npg\__init__.py", line 98, in <module>
    from . import npg_setops, npg_io, npg_table, npg_create, npg_utils
  File "C:\Users\*******\Documents\ArcGIS\SDK\numpy_geometry-master\arcpro_npg\npg\npg\npg_create.py", line 89, in <module>
    from npg_helpers import _to_lists_
ModuleNotFoundError: No module named 'npg_helpers'
 Failed to execute (RotateFeatures).
#####################

It seems I do have the npg_helpers.py file in the right folder, so I'm not sure what to make of this. 

Thank you again, and please forgive a new user for any mistakes!
Chad

**Comment by Dan-Patterson** [comment]
Chad, I will try to have a look at it tonight to make sure everything is there.  In the interim could you move the tools to a simpler folder like
C:\arc_tools\numpy_geometry_master

From: cjchristensen8 ***@***.***>
Sent: Thursday, June 15, 2023 3:46 PM
To: Dan-Patterson/numpy_geometry ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [Dan-Patterson/numpy_geometry] Error message on npg_helpers (Discussion #2)


hi Dan - a huge thanks for craeting this toolset, and i offer my great admiration for your help to the community.
When I tried the tool to rotate objects, I got this error string (using ArcGIS Pro 2.5 on a Windows 10 machine):

#########################
Traceback (most recent call last):
File "C:\Users*\Documents\ArcGIS\SDK\numpy_geometry-master\arcpro_npg\npg\tbx_tools.py", line 131, in
from npg.npg_create import circle
File "C:\Users*\Documents\ArcGIS\SDK\numpy_geometry-master\arcpro_npg\npg\npg_init_.py", line 98, in
from . import npg_setops, npg_io, npg_table, npg_create, npg_utils
File "C:\Users*******\Documents\ArcGIS\SDK\numpy_geometry-master\arcpro_npg\npg\npg\npg_create.py", line 89, in
from npg_helpers import to_lists
ModuleNotFoundError: No module named 'npg_helpers'
Failed to execute (RotateFeatures).
#####################

It seems I do have the npg_helpers.py file in the right folder, so I'm not sure what to make of this.

Thank you again, and please forgive a new user for any mistakes!
Chad

-

**Reply by cjchristensen8** [reply]
Thank you - Dan.  I moved it to a folder called C:\MapWork\ArcGIS\Toolbox and got the same results.  I agree that the original path was worth shortening!  I appreciate your work, and I don't want to suggest I have an urgent need - I was able to work around the problem layer with some time and manual operation, but I would love to have the tools for future use.  
Happy bug hunting--
Chad
